Stack-Calculator.Tests/StackCalculatorTests.cs

```diff
@@ -0,0 +1,214 @@
+
+public class StackCalculatorTests
+{
+    [TestCaseSource(nameof(Stacks))]
+    public void Calculate_WithTooManyNumbers_ShouldThrowArgumentException(IStack stack)
+    {
+        var stackCalculator = new StackCalculator(stack);
+
+        Assert.Throws<ArgumentException>(() => stackCalculator.Calculate("5 5 5 5 +"));
+    }
+
+    [TestCaseSource(nameof(Stacks))]
+    public void Calculate_WithTooManyOperations_ShouldThrowArgumentException(IStack stack)
+    {
+        var calculator = new StackCalculator(stack);
+
+        Assert.Throws<ArgumentException>(() => calculator.Calculate("5 + + +"));
+    }
+
+    [TestCaseSource(nameof(Stacks))]
+    public void Calculate_WithDivisionByZeroExpression_ShouldThrowDivideByZeroException(IStack stack)
+    {
+        var calculator = new StackCalculator(stack);
+
+        Assert.Throws<DivideByZeroException>(() => calculator.Calculate("0 2 /"));
+    }
+
+    [TestCaseSource(nameof(Stacks))]
+    public void Calculate_WithUsingOfInvalidOperation_ShouldThrowArgumentException(IStack stack)
+    {
+        var calculator = new StackCalculator(stack);
+
+        Assert.Throws<ArgumentException>(() => calculator.Calculate(" 5 5 %"));
+    }
+
+    [TestCaseSource(nameof(Stacks))]
+    public void Calculate_WithUsingOfInvalidSymbols_ShouldThrowArgumentException(IStack stack)
+    {
+        var calculator = new StackCalculator(stack);
+
+        Assert.Throws<ArgumentException>(() => calculator.Calculate(" 5 f 5 + +"));
+    }
+
+    [TestCaseSource(nameof(Stacks))]
+    public void Calculate_WithEmptyString_ShouldThrowArgumentExeption(IStack stack)
+    {
+        var calculator = new StackCalculator(stack);
+
+        Assert.Throws<ArgumentException>(() => calculator.Calculate(string.Empty));
+    }
+
+    [TestCaseSource(nameof(StacksForExpressionWithAdditionTest))]
+    public void Calculate_WithAdditionExpression_ShouldReturnExpectedResult(IStack stack, string data, float expectedResult)
+    {
+        var calculator = new StackCalculator(stack);
+        float result;
+
+        result = calculator.Calculate(data);
+
+        Assert.That(expectedResult, Is.EqualTo(result));
+    }
+
+    [TestCaseSource(nameof(StacksForExpressionWithSubstractionTest))]
+    public void Calculate_WithSubstractionExpression_ShouldReturnExpectedResult(IStack stack, string expression, float expectedResult)
+    {
+        var calculator = new StackCalculator(stack);
+        float result;
+
+        result = calculator.Calculate(expression);
+
+        Assert.That(expectedResult, Is.EqualTo(result));
+    }
+
+    [TestCaseSource(nameof(StacksForExpressionWithMultiplicationTest))]
+    public void Calculate_WithMultiplicationExpression_ShouldReturnExpectedResult(IStack stack, string expression, double expectedResult)
+    {
+        var calculator = new StackCalculator(stack);
+        float result;
+
+        result = calculator.Calculate(expression);
+
+        Assert.That(expectedResult, Is.EqualTo(result));
+    }
```

Comment:
И так можно со всеми :)
```suggestion
    [TestCaseSource(nameof(StacksForExpressionWithSubstractionTest))]
    [TestCaseSource(nameof(StacksForExpressionWithMultiplicationTest))]
    public void Calculate_WithMultiplicationExpression_ShouldReturnExpectedResult(IStack stack, string expression, double expectedResult)
    {
        var calculator = new StackCalculator(stack);
        float result;

        result = calculator.Calculate(expression);

        Assert.That(expectedResult, Is.EqualTo(result));
    }
```